src/fullnode.rs

```diff
@@ -285,19 +160,24 @@ impl Fullnode {
     ///   `--------`  |                               |    `------------`
     ///               `-------------------------------`
     /// ```
-    pub fn new_validator(
+    pub fn new_with_bank(
         keypair: Keypair,
         bank: Bank,
         entry_height: u64,
         ledger_tail: &[Entry],
-        node: TestNode,
-        entry_point: &NodeInfo,
+        mut node: TestNode,
+        leader_info: Option<&NodeInfo>,
         exit: Arc<AtomicBool>,
         ledger_path: Option<&str>,
-        _sigverify_disabled: bool,
+        sigverify_disabled: bool,
     ) -> Self {
+        if leader_info.is_none() {
+            node.data.leader_id = node.data.id;
```

Comment:
I think this is redundant with https://github.com/solana-labs/solana/blob/cabe80b129a6303c8824caaafa8ffad839b9b943/src/bin/fullnode.rs#L88